rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,149 @@
+| renderExample.tsx   | example.view.tsx    | `export const renderExample = () => {..}`                                           | `.render` might work as well, to match symbol name.                                                                                                                                                                                                                                                                       |
+| useExample.ts       | example.hook.ts     | `export const useExample = () => {}`                                                | main logic encapsulated under hook -> name should be the same as encapsulated component (in our case `example`)                                                                                                                                                                                                           |
+| useExampleStyles.ts | example.styles.ts   | `export const useStyles = makeStyles(); export const useIconStyles = makeStyles();` | with `makeStyles` 1 to N hooks for particular component styling needs. Why we currently use a solution that provides hooks, having `.styles` could be a good enough convention to explicitly communicate the intent. (if we switch implementation in the future we wouldn't need to change the type name as it's generic) |
+| Example.types.ts    | example.types.ts    | `export type Foo = {}; export interface Hello {}`                                   | aggregated set of reusable type definitions used within other parts of component/also might be re-exported for public API surface                                                                                                                                                                                         |
+
+### Explanation
+
+- use consistent names for all files.
+- follow a pattern that describes the symbol's feature or the file scope(component name) then its type.
+  - The recommended pattern is `feature.type.ts`.
+- use conventional **type** names including
+  - `.hook`
+  - `.view`
+  - `.styles`
+  - `.types`
+  - `.stories`
+  - `.utils`
+  - invent additional type names if you must but take care not to create too many
+
+**Why?**
+
+- Type names provide a consistent way to quickly identify what is in the file.
+- Type names make it easy to find a specific file type using an editor or IDE's fuzzy search techniques.
+- Type names provide pattern matching for any automated tasks.
+
+---
+
+- use `.tsx` only for files that contain view/rendering related logic
```

Comment:
```suggestion
- use `.tsx` only for files that contain view/rendering related logic
  - enable lint rule that will enforce one way 
```

I suggest to add into proposal an ESLint rule for this, otherwise it's easy to fail this convention. There is [jsx-filename-extension](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md), but I have not seen a similar for TS files:
- no JSX in files - use `.ts`
- JSX in files - use `.tsx` (this is forced by compiler anyway)